rust_snuba/rust_arroyo/src/processing/mod.rs

```diff
@@ -104,9 +104,9 @@ pub struct StreamProcessor<'a, TPayload: Clone> {
     metrics_buffer: metrics_buffer::MetricsBuffer,
 }
 
-impl<'a, TPayload: 'static + Clone> StreamProcessor<'a, TPayload> {
+impl<TPayload: 'static + Clone> StreamProcessor<TPayload> {
     pub fn new(
-        consumer: Box<dyn Consumer<'a, TPayload> + 'a>,
+        consumer: Arc<Mutex<dyn Consumer<TPayload>>>,
```

Comment:
Not sure I get why you need the consumer to be in a Mutex. Are we using from multiple threads ?

Reply:
Arc is immutable by itself, and the consumer needs to be mut. We are going to pass this into the consumer context to use during rebalancing callbacks - this is the part requires requires thread safety (and is the motivation to make this Arc)

Reply:
ok